### ISSUE TITLE
crash with GDTCCTUploader

### ISSUE THREAD

**Issue by Jubian** [firebase/firebase-ios-sdk#3530]
### [REQUIRED] Step 2: Describe your environment

  * Xcode version: 10.2.1
  * Firebase SDK version: 6.6.0
  * Firebase Component: GoogleDataTransportCCTSupport
  * Component version: 1.0.0

### [REQUIRED] Step 3: Describe the problem
open the application for a while, it crashed.I don't know what caused the crash
with stack trace:
```
libsystem_kernel.dylib`__pthread_kill:
    0x1eda410fc <+0>:  mov    x16, #0x148
    0x1eda41100 <+4>:  svc    #0x80
->  0x1eda41104 <+8>:  b.lo   0x1eda4111c               ; <+32>
    0x1eda41108 <+12>: stp    x29, x30, [sp, #-0x10]!
    0x1eda4110c <+16>: mov    x29, sp
    0x1eda41110 <+20>: bl     0x1eda36800               ; cerror_nocancel
    0x1eda41114 <+24>: mov    sp, x29
    0x1eda41118 <+28>: ldp    x29, x30, [sp], #0x10
    0x1eda4111c <+32>: ret    

```
crash happen at:
```
id completionHandler =
        ^(NSData *_Nullable data, NSURLResponse *_Nullable response, NSError *_Nullable error) {
          NSAssert(!error, @"There should be no errors uploading events: %@", error);
          NSError *decodingError;
          gdt_cct_LogResponse logResponse = GDTCCTDecodeLogResponse(data, &decodingError);
          if (!decodingError && logResponse.has_next_request_wait_millis) {
            self->_nextUploadTime =
                [GDTClock clockSnapshotInTheFuture:logResponse.next_request_wait_millis];
          } else {
            // 15 minutes from now.
            self->_nextUploadTime = [GDTClock clockSnapshotInTheFuture:15 * 60 * 1000];
          }
          pb_release(gdt_cct_LogResponse_fields, &logResponse);
          [package completeDelivery];
          if (self->_backgroundID != GDTBackgroundIdentifierInvalid) {
            [[GDTApplication sharedApplication] endBackgroundTask:self->_backgroundID];
            self->_backgroundID = GDTBackgroundIdentifierInvalid;
          }
          self.currentTask = nil;
          self.currentUploadPackage = nil;
        };
```
and the print information
```
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'There should be no errors uploading events: Error Domain=NSURLErrorDomain Code=-1005 "The network connection was lost." UserInfo={NSUnderlyingError=0x282325950 {Error Domain=kCFErrorDomainCFNetwork Code=-1005 "(null)" UserInfo={_kCFStreamErrorCodeKey=54, _kCFStreamErrorDomainKey=1}}, NSErrorFailingURLStringKey=https://firebaselogging.googleapis.com/v0cc/log/batch, NSErrorFailingURLKey=https://firebaselogging.googleapis.com/v0cc/log/batch, _kCFStreamErrorDomainKey=1, _kCFStreamErrorCodeKey=54, NSLocalizedDescription=The network connection was lost.}'
*** First throw call stack:
(0x1edeabef8 0x1ed079a40 0x1eddc006c 0x1ee8ac3e0 0x10176654c 0x1ee4847bc 0x1ee498f5c 0x1ee937b6c 0x1ee83fcc8 0x1ee83f19c 0x1ee939a40 0x103537840 0x103538de4 0x10353c1e0 0x10353b6b8 0x10354a1f8 0x10354abe4 0x1edac6190 0x1edac8d00)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by mikehaney24** [comment]
Thanks for letting us know! This is happening because if a connection fails for any reason, the first NSAssert statement in the completion handler will trigger. It's most likely not a real issue, but I'll discuss removing asserts with the team so the dev experience is a bit better.

**Comment by Nathan1258** [comment]
Go [here ](https://stackoverflow.com/questions/57411528/why-does-my-swift-app-crash-whenever-i-go-offline?noredirect=1#comment101304964_57411528)for more context with someone with more knowledge with me explaining why it's horrible code.

**Comment by puf** [comment]
Another report here: https://stackoverflow.com/q/57411528

**Comment by paulb777** [comment]
We'll investigate getting a patch out for this soon.

**Comment by Nathan1258** [comment]
> We'll investigate getting a patch out for this soon.

Please do, I cannot do anything with my developments until this is fixed as it's always crashing my app.

**Comment by paulb777** [comment]
You may be able to workaround by deleting the NSAssert. 

Also going back to Firebase 6.5.0 would avoid the issue.

**Comment by mikehaney24** [comment]
Another solution is compile with NS_BLOCK_ASSERTIONS defined--this is the default in Release builds because assertions are meant to get your attention when things go wrong during development.

I'm currently working on a PR to remove all asserts.

**Comment by mikehaney24** [comment]
This should be addressed by #3535, we're working on creating a new release that contains these changes now.

**Comment by paulb777** [comment]
The fix is now released.

Run `pod update` to get  GoogleDataTransport (1.1.0), GoogleDataTransportCCTSupport (1.0.1), GoogleDataTransportCCTSupport (1.0.1)
